packages/core/core-flows/src/defaults/steps/create-default-region.ts

```diff
@@ -0,0 +1,43 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { CreateRegionDTO, IRegionModuleService } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+import { createRegionsWorkflow } from "../../region/workflows/create-regions"
+
+type CreateDefaultRegionStepInput = {
+  region: CreateRegionDTO
+}
+
+export const createDefaultRegionStepId = "create-default-region"
+export const createDefaultRegionStep = createStep(
+  createDefaultRegionStepId,
+  async (data: CreateDefaultRegionStepInput, { container }) => {
+    const service = container.resolve<IRegionModuleService>(
+      ModuleRegistrationName.REGION
+    )
+
+    let shouldDelete = false
+    let [region] = await service.list({}, { take: 1 })
+
+    if (!region) {
+      const { result } = await createRegionsWorkflow(container).run({
+        input: { regions: [data.region] },
+      })
+
+      region = result[0]
+      shouldDelete = true
```

Comment:
I think instead of using this flag, what we can do is return 

```
return new StepResponse(region, { regionId: newlyCreatedRegion.id | undefined })
```

And in the reverse step we check if `regionId` is set (it will be only in the case of a newly created region)